mm2src/mm2_main/src/lp_ordermatch/orderbook_rpc.rs

```diff
@@ -111,6 +111,38 @@ pub fn is_my_order(my_pub: &Option<String>, order_pubkey: &str) -> bool {
     my_pub.as_ref().map(|my| my == order_pubkey).unwrap_or(false)
 }
 
+// ZHTLC protocol coin uses random keypair to sign P2P messages per every order.
+// So, each ZHTLC order has unique «pubkey» field that doesn’t match node persistent pubkey derived from passphrase.
+// We can compare pubkeys from maker_orders and from asks or bids, to find our order.
+pub fn is_mine_zhtlc(my_orders_pubkeys: &Vec<String>, my_pub: &Option<String>, order_pubkey: &str) -> bool {
+    let mut is_mine_zhtlc = false;
+    for my_pubkey in my_orders_pubkeys {
+        if my_pubkey == order_pubkey {
+            is_mine_zhtlc = true;
+            break;
+        }
+    }
+    drop_mutability!(is_mine_zhtlc);
+    let is_my_order = is_my_order(my_pub, order_pubkey);
+    is_mine_zhtlc || is_my_order
+}
+
+async fn get_my_orders_pubkeys(ctx: &MmArc) -> Vec<String> {
+    let ordermatch_ctx = OrdermatchContext::from_ctx(ctx).expect("from_ctx failed");
+    let my_orders = ordermatch_ctx.maker_orders_ctx.lock().orders.clone();
+    let mut res = Vec::new();
+    for (_, order_mutex) in my_orders {
```

Comment:
Iterating over all node's orders on each orderbook request might cause performance issues or even deadlocks. It seems that we need to create `my_p2p_pubkeys: HashSet<String>` field in OrdermatchContext and add each pubkey of the random `p2p_privkey` upon order creation. The pubkey should be removed when the order is deleted.

Reply:
done
this method was removed, new field `my_p2p_pubkeys` was added